### ISSUE TITLE
learning rate can't update in resuming training with hovorod

### ISSUE THREAD

**Issue by TLedTS** [PyTorchLightning/pytorch-lightning#12065]
## 🐛 Bug

When I resumed training the model with `LearningRateMonitor`, the LR written in the log won't update anymore but just keep the same with original status of the ckpt file.

### To Reproduce

```
import os
import torch
from torch import nn
import torch.nn.functional as F
from torchvision.datasets import MNIST
from torch.utils.data import DataLoader, random_split
from torchvision import transforms
import pytorch_lightning as pl
from pytorch_lightning.callbacks import LearningRateMonitor
import horovod.torch as hvd

class LitAutoEncoder(pl.LightningModule):
    def __init__(self):
        super().__init__()
        self.encoder = nn.Sequential(nn.Linear(28 * 28, 128), nn.ReLU(), nn.Linear(128, 3))
        self.decoder = nn.Sequential(nn.Linear(3, 128), nn.ReLU(), nn.Linear(128, 28 * 28))

    def forward(self, x):
        # in lightning, forward defines the prediction/inference actions
        embedding = self.encoder(x)
        return embedding

    def training_step(self, batch, batch_idx):
        # training_step defines the train loop. It is independent of forward
        x, y = batch
        x = x.view(x.size(0), -1)
        z = self.encoder(x)
        x_hat = self.decoder(z)
        loss = F.mse_loss(x_hat, x)
        self.log("train_loss", loss)
        self.log("lr", self.optimizer.param_groups[0]['lr'])
        return loss

    def configure_optimizers(self):
        self.optimizer = torch.optim.Adam(self.parameters(), lr=0.01)
        # self.optimizer = hvd.optimizer.DistributedOptimizer(self.optimizer)
        lr_scheduler = torch.optim.lr_scheduler.CosineAnnealingWarmRestarts(optimizer=self.optimizer, T_0=10, T_mult=2, eta_min=0)
        return [self.optimizer], [lr_scheduler]


def main():
    dataset = MNIST(os.getcwd(), download=True, transform=transforms.ToTensor())
    train, val = random_split(dataset, [55000, 5000])

    autoencoder = LitAutoEncoder()
    lr_monitor = LearningRateMonitor(logging_interval='step')
    trainer = pl.Trainer(
        gpus=1,
        max_epochs=64,
        strategy='horovod',
        callbacks=[lr_monitor],
        limit_train_batches=32
        )
    trainer.fit(autoencoder, DataLoader(train), DataLoader(val), ckpt_path='lightning_logs/version_14/checkpoints/epoch=4-step=63.ckpt')


if __name__ == "__main__":
    main()
```

### Expected behavior

Use the MNIST as an example, train it from scratch and save a model in middle.
Then pass the model path to trainer to resume train the model.
Now check the log in tensorboard, you can see the lr-Adam won't update.
But for the 'lr' we write in log , it is normal.
And it will be normal when the strategy is 'ddp' or with the **version 1.4.2** of pytorch_lightning.
![image](https://user-images.githubusercontent.com/27993525/155298550-44ef15f3-b8af-4368-842a-73e3255640f5.png)
If anyone has any suggestion, please kindly let me know! Thanks a lot!

### Environment

- PyTorch Lightning Version (e.g., 1.5.0):1.5.4
- PyTorch Version (e.g., 1.10):1.10.0
- Python version (e.g., 3.9):3.7.11
- OS (e.g., Linux):Linux
- Horovod version:0.23.0
- MPI version:3.0.0
- CUDA version:11.1
- NCCL version:2.7.8

### Additional context
but I found that if I use the old way to define the distributed optimizer by uncomment the line below, then it will update the learning rate normally.
```python
def configure_optimizers(self):
        self.optimizer = torch.optim.Adam(self.parameters(), lr=0.01)
        # self.optimizer = hvd.optimizer.DistributedOptimizer(self.optimizer)
```

**Comment by amorehead** [comment]
I am currently seeing a similar issue with my LightningModule w/ LearningRateCallback. Using `torch.optim.Adam` and any optimizer, I am unable to see my learning rate plotted or changed in any way during training.

`
def configure_optimizers(self):
        """Called to configure the trainer's optimizer(s)."""
        optimizer = Adam(self.parameters(), lr=self.lr, weight_decay=self.weight_decay)
        scheduler = construct_scheduler(optimizer, self.scheduler_type, self.lr, self.warmup, self.num_epochs)
        return {
            'optimizer': optimizer,
            'lr_scheduler': {
                'scheduler': scheduler,
                'monitor': self.metric_to_track,
                'frequency': self.trainer.check_val_every_n_epoch
            }
        }
`

**Comment by rohitgr7** [comment]
> I am currently seeing a similar issue with my LightningModule w/ LearningRateCallback. Using torch.optim.Adam and any learning rate scheduler, I am unable to see my learning rate plotted or changed in any way during training.

@amorehead with horovod?

**Comment by amorehead** [comment]
@rohitgr7,

Actually, with DDP. I found that the issue only showed up when I returned a dictionary containing my `optimizer` and `lr_scheduler` keys.

**Comment by ysyyork** [comment]
Any progress here? same issue

**Comment by rohitgr7** [comment]
here:
```py
    def configure_optimizers(self):
        self.optimizer = torch.optim.Adam(self.parameters(), lr=0.01)
        # self.optimizer = hvd.optimizer.DistributedOptimizer(self.optimizer)
        lr_scheduler = torch.optim.lr_scheduler.CosineAnnealingWarmRestarts(optimizer=self.optimizer, T_0=10, T_mult=2, eta_min=0)
        return [self.optimizer], [lr_scheduler]
```
the scheduler is configured with interval='epoch' by default. ref: https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers

and learning rate monitor is by `step`.

```py
lr_monitor = LearningRateMonitor(logging_interval='step')
```

Possibly that's why its not updating the lr and lr monitor just shows the same lr for that epoch in all the steps.

can anyone share a reproducible script using: https://colab.research.google.com/github/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report/bug_report_model.ipynb

thank you!
